File: Algorithms/011.ContainerWithMostWater.java
public class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int maxArea = 0;
        int area=0, i=0, j=n-1;
        while (i < j) {
            if (height[i] < height[j]) {
                area = height[i] * (j-i);
                i++;
            } else {
                area = height[j] * (j-i);
                j--;
            }
            if (area > maxArea) {
                maxArea = area;
            }
        }
        return maxArea;
    }
}

File: Algorithms/011.ContainerWithMostWater.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        i, j = 0, len(height)-1
        max_area = 0
        while i < j:
            if height[i] < height[j]:
                area = (j-i) * height[i]
                i += 1
            else:
                area = (j-i) * height[j]
                j -= 1
            max_area = max(area, max_area)
        return max_area
